PatientRegistrationApp/AddPatientDialog.cs

```
@@ -35,26 +33,37 @@ private void btnCancel_Click(object sender, EventArgs e)
-            currSheet.Cells[currRow, kAddressLoc].Value = textAddress.Text;
-            currSheet.Cells[currRow , kCityLoc].Value = textCity.Text;
-            currSheet.Cells[currRow, kStateLoc].Value = textState.Text;
-            currSheet.Cells[currRow, kZipLoc].Value = textZip.Text;
-            currSheet.Cells[currRow, kHomePhoneLoc].Value = textHomePhone.Text;
-            currSheet.Cells[currRow, kCellPhoneLoc].Value = textCellPhone.Text;
-            currSheet.Cells[currRow, kReturnDateLoc].Value = textReturnDate.Text;
-
-            m_existingWkBook.Save(); // save this newly added row
+            try
+            {
+                using (var workbook = new XLWorkbook(m_filePath))
+                {
+                    var worksheet = workbook.Worksheet("Patient_Registration MASTER");
+
+                    int lastRow = worksheet.LastRowUsed()?.RowNumber() ?? 0;
+                    int currRow = lastRow + 1;
+
+                    worksheet.Cell(currRow, kCurrDateLoc).Value = DateTime.Today.ToShortDateString();
+                    worksheet.Cell(currRow, kFirstNameLoc).Value = textFirstName.Text;
+                    worksheet.Cell(currRow, kLastNameLoc).Value = textLastName.Text;
+                    worksheet.Cell(currRow, kAddressLoc).Value = textAddress.Text;
+                    worksheet.Cell(currRow, kCityLoc).Value = textCity.Text;
+                    worksheet.Cell(currRow, kStateLoc).Value = textState.Text;
+                    worksheet.Cell(currRow, kZipLoc).Value = textZip.Text;
+                    worksheet.Cell(currRow, kHomePhoneLoc).Value = textHomePhone.Text;
+                    worksheet.Cell(currRow, kCellPhoneLoc).Value = textCellPhone.Text;
+
+                    if (DateTime.TryParse(textReturnDate.Text, out DateTime parsedDate))
+                    {
+                        worksheet.Cell(currRow, kReturnDateLoc).Value = parsedDate; // Saves as a true Date
+                    }
```

change request: I think if this if statement fails, we should throw an exception and the catch block should show an error to the user. Something like `"Cannot parse the specified date for row {x}. Please report an issue to the developers immediately"`. We don't want to save on a workbook that contains partial data. We either want to have a full success or a full failure. Anything in between will probably result in data consistency issues down the road and we should prevent that here before going forward.